api/v1alpha2/kubeadmbootstrapconfig_types.go

```diff
@@ -36,6 +36,9 @@ type KubeadmConfigSpec struct {
 	// AdditionalUserDataFiles specifies extra files to be passed to user_data upon creation.
 	// +optional
 	AdditionalUserDataFiles []Files `json:"additionalUserDataFiles,omitempty"`
+	// AdditionalCommands specifies extra commands to be passed to user_data upon creation
+	// +optional
+	AdditionalCommands []string `json:"additionalCommands,omitempty"`
```

Comment:
I know previously we had discussed having both pre and post commands rather than just a single AdditionalCommands field. Could you update this for that? Maybe `AdditionalPreCommands` and `AdditionalPostCommands`?

Updating this file will also require updating the generated CRDs.

Reply:
Supporting `AdditionalPreCommands`  and `AdditionalPostCommands` will also need changes to the cloud-init template, correct?

Something like this:
```
runcmd:
{{- template "commands" .AdditionalPreCommands }}
  - 'kubeadm init --config /tmp/kubeadm.yaml'
{{- template "commands" .AdditionalPostCommands }}
```

Reply:
That would be correct, yes.

Reply:
@detiber I was wondering if there an automatic way to generate the updated YAML using the updated type or maybe the reverse, generate the type using the YAML?

Reply:
if you modify the struct and run make generate the yaml should all get updated (not the cloud-init stuff though, that will be manual)